### ISSUE TITLE
Acceptance test asserts and exceptions

### ISSUE THREAD

**Issue by phil-davis** [owncloud/core#36810]
Currently we fail acceptance tests by either throwing an exception or using an `Assert::`

Throwing and exception produces a big/long/red traceback.

Using `Assert::` gives a nice short message about what was the problem with the assert. But sometimes the message is something like "failed asserting that false is true", which is not helpful for someone trying to understand what really failed.

Improve this by:

- [ ] On every `Assert::*` call, put something in the message parameter that will give a better description of what was being tested and what failed.
- [ ] `Given` steps are not expected to be tests of the system. Throw exceptions when checking for errors in `Given` steps. That will produce a full traceback for a developer to quickly find exactly where in the code it crashed.
- [ ] `When` steps should just do actions and not check anything. If you really want/need to check some condition in a `When` step then throw an exception (for similar reasons to `Given`)
- [ ] `Then` steps should always use `Assert::*` - change any exceptions thrown in `Then` steps to be `Assert::*` with suitable message

Break these rules when appropriate (but not too often)

**Comment by phil-davis** [comment]
- [ ] also check that the order of parameters in `Assert::assert*` is around the right way - expected then actual. (Otherwise the "automatic" message that is emitted when the Assert fails will be confusing.

(e.g. see PR #36938 where I fixed some of these that I noticed)

- [ ] when adjusting these, think about renaming the variables so they are like `$expectedText` `$actualText` - using `expected` and `actual` in the variable names might help the code to be more readable.
